ajax_with_jquery_exercise/nacker_hews_AJAX/app.js

```diff
@@ -0,0 +1,292 @@
+  let $topiclist = $(".topiclist");
+  let $favorite = $(".favoritetoggle");
+  let $all = $(".alltoggle");
+  let $topStories = $(".top-stories");
+  let $bestStories = $(".best-stories");
+  let $newStories = $(".new-stories");
+  let $login = $(".logtoggle");
+  let $logout = $(".logout");
+  let $loginForm = $(".log-in");
+  let $signUpForm = $(".sign-up");
+  let storage = [];
+
+  let hackerNews = [];
+
+  //Function to fetch top 25 results for fetch news.
+  function fetchTopNews() {
+    $topiclist.empty();
+    $.get("https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty")
+    .then(function(res){
+      hackerNews = hackerNews.concat(res.slice(0,25));
+      hackerNews.forEach(function(value) {
+        $.get("https://hacker-news.firebaseio.com/v0/item/" + value + ".json?print=pretty")
+        .then(function(res){
+          console.log(res);
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star-empty hide-all",
+          })
+          $star.data("story_id",res.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: res.url,
+            text: res.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${res.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          hackerNews = [];
+          //Add star if auto logged in
+          if (localStorage.getItem("Nacker")) {
+            $(".glyphicon").removeClass("hide-all");    
+          }
+        })
+      })
+    })
+  } 
+  //Function to fetch best 25 results for fetch news.
+  function fetchBestNews() {
+    $topiclist.empty();
+    $.get("https://hacker-news.firebaseio.com/v0/beststories.json?print=pretty")
+    .then(function(res){
+      hackerNews = hackerNews.concat(res.slice(0,25));
+      hackerNews.forEach(function(value) {
+        $.get("https://hacker-news.firebaseio.com/v0/item/" + value + ".json?print=pretty")
+        .then(function(res){
+          console.log(res);
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star-empty hide-all",
+          })
+          $star.data("story_id",res.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: res.url,
+            text: res.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${res.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          hackerNews = [];
+          //Add star if auto logged in
+          if (localStorage.getItem("Nacker")) {
+            $(".glyphicon").removeClass("hide-all");    
+          }
+        })
+      })
+    })
+  } 
+  //Function to fetch new 25 results for fetch news.
+  function fetchNewNews() {
```

Comment:
It looks like there's a fair amount of code duplication between `fetchTopNews`, `fetchBestNews`, and `fetchNewNews`. If you're looking to refactor, I'd try refactoring these into a single function that takes an argument or two.